jobs/jobs.go

```diff
@@ -43,14 +86,45 @@ func (mgr *JobManager) AddJobs(jobs ...Job) {
 func (mgr *JobManager) AddJob(j Job) {
 	mgr.wgJobs.Add(1)
 	go func() {
-		defer mgr.wgJobs.Done()
-		j(mgr.errChan)
+		if err := mgr.jobCapacity.Acquire(mgr.ctx, 1); err != nil {
+			mgr.tryCacheError(err)
+			mgr.jobCapacity.Release(1)
+			return
+		}
+		go func() {
+			defer mgr.wgJobs.Done()
```

Comment:
Do we  always want to require that .Done and .Release are called together by wrapping them in a helper?

Reply:
they would still be called one after the other

Reply:
right, I thought it might help codify the need to do them in tangent, but in cases like failing to acquire the job capacity, we don't have anything to release